src/tlo/dependencies.py

```diff
@@ -122,27 +126,33 @@ def depth_first_search(module):
             )
             for dependency in sorted(dependencies):
                 if dependency not in module_instance_map:
-                    alternatives_with_instances = [
-                        name for name, instance in module_instance_map.items()
-                        if dependency in instance.ALTERNATIVE_TO
-                    ]
-                    if len(alternatives_with_instances) != 1:
-                        message = (
-                            f'Module {module} depends on {dependency} which is '
-                            'missing from modules to register'
-                        )
-                        if len(alternatives_with_instances) == 0:
-                            message += f' as are any alternatives to {dependency}.'
-                        else:
-                            message += (
-                                ' and there are multiple alternatives '
-                                f'({alternatives_with_instances}) so which '
-                                'to use to resolve dependency is ambiguous.'
+                    if auto_register_modules:
+                        # add missing dependencies and associated classes in module instance map dictionary
+                        module_class = get_module_class_map(set())[dependency](resourcefilepath=data_folder)
+                        module_instance_map.update({dependency: module_class})
```

Comment:
On the third point, the module `get_module_class_map ` is called once in the `for dependency in sorted(dependencies):` and that is done within :

```
                if dependency not in module_instance_map:
                    if auto_register_modules:
```
This makes its not redundant. It would have been redundant if it was placed just above if statement. What do you think @matt-graham @mnjowe

Reply:
Hi @jkumwenda. Even inside that if statement it is still redundant because it will keep getting initialised at every dependecy in the for loop that's not in the module instance map. @matt-graham is right, we need to initialise that outside the for loop and re-use it within that it statement block.

Reply:
Ok, thank you for clarification, We will resolve that and do a quick debug before committing.